### PR TITLE
fix: catch UnicodeDecodeError in `_extract_session_name` (#346)

### LINKED ISSUE #346: [aw][code health] `_extract_session_name` crashes on non-UTF-8 `plan.md` + missing test coverage for edge cases
## Root Cause

`_extract_session_name` (`parser.py:160`) calls `plan.read_text(encoding="utf-8")` but only catches `OSError`:

```python
try:
    first_line = plan.read_text(encoding="utf-8").split("\n", maxsplit=1)[0]
    ...
except OSError as exc:
    logger.debug("Could not read session name from {}: {}", plan, exc)
```

`UnicodeDecodeError` is **not** a subclass of `OSError`, so a `plan.md` containing non-UTF-8 bytes (e.g., a Latin-1 encoded file, a binary accidentally placed in the session directory) raises an unhandled exception. This propagates up through `build_session_summary` → `get_all_sessions` and crashes the entire `summary`, `cost`, and `live` commands.

At the same time, a survey of the test suite reveals several related edge cases that have no coverage:

| Gap | What's missing |
|-----|----------------|
| `_extract_session_name` — non-UTF-8 `plan.md` | crash scenario above |
| `_extract_session_name` — `plan.md` with no H1 header | graceful None return untested |
| `_extract_session_name` — empty H1 (`# \n`) | already handled, but no test |
| `_total_output_tokens` — empty `model_metrics` + non-zero `active_output_tokens` | third branch never hit by a direct test |
| `_filter_sessions` — session with `start_time=None` | always excluded, never tested |

### 1. Catch `UnicodeDecodeError` in `_extract_session_name`

```python
# parser.py
except (OSError, UnicodeDecodeError) as exc:
    logger.debug("Could not read session name from {}: {}", plan, exc)
```

### 2. Add missing unit tests

**`tests/copilot_usage/test_parser.py`** — add to the `_extract_session_name` section:

```python
def test_extract_session_name_unicode_decode_error(self, tmp_path: Path) -> None:
    plan = tmp_path / "plan.md"
    plan.write_bytes(b"\xff\xfe not valid utf-8")
    assert _extract_session_name(tmp_path) is None  # was crashing before fix

def test_extract_session_name_no_h1_header(self, tmp_path: Path) -> None:
    (tmp_path / "plan.md").write_text("## Subheading\nsome text\n", encoding="utf-8")
    assert _extract_session_name(tmp_path) is None

def test_extract_session_name_empty_h1(self, tmp_path: Path) -> None:
    (tmp_path / "plan.md").write_text("# \n", encoding="utf-8")
    assert _extract_session_name(tmp_path) is None
```

**`tests/copilot_usage/test_report.py`** — add direct coverage for `_total_output_tokens`:

```python
def test_total_output_tokens_empty_metrics_includes_active_tokens(self) -> None:
    """Empty model_metrics + active tokens → returns active tokens."""
    session = SessionSummary(
        session_id="test",
        model_metrics={},
        active_output_tokens=500,
        is_active=True,
    )
    assert _total_output_tokens(session) == 500

def test_total_output_tokens_pure_active_no_double_count(self) -> None:
    """Pure-active session: active tokens already in model_metrics; no double-count."""
    session = SessionSummary(
        session_id="test",
        model_metrics={
            "claude": ModelMetrics(
                requests=RequestMetrics(count=0),
                usage=TokenUsage(outputTokens=350),
            )
        },
        active_output_tokens=350,
        is_active=True,
    )
    # has_shutdown_metrics is False (count=0), so baseline only
    assert _total_output_tokens(session) == 350
```

**`tests/copilot_usage/test_report.py`** — `_filter_sessions` with `start_time=None`:

```python
def test_filter_sessions_excludes_session_with_no_start_time(self) -> None:
    since = datetime(2026, 1, 1, tzinfo=UTC)
    session = SessionSummary(session_id="s", start_time=None)
    assert _filter_sessions([session], since=since, until=None) == []
```

> Generated by [Code Health Analysis](https://github.com/microsasa/cli-tools/actions/runs/23508596425) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Fcode-health%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
## Summary

Fixes #346 — `_extract_session_name` crashes on non-UTF-8 `plan.md` files because `UnicodeDecodeError` is not a subclass of `OSError`.

### Bug fix (`src/copilot_usage/parser.py`)
- Broadened the `except` clause in `_extract_session_name` from `OSError` to `(OSError, UnicodeDecodeError)` so that binary or non-UTF-8 `plan.md` files are gracefully handled (returns `None` + debug log) instead of propagating an unhandled exception.

### New tests

**`tests/copilot_usage/test_parser.py`** — 3 new tests in `TestExtractSessionName`:
| Test | Covers |
|------|--------|
| `test_extract_session_name_unicode_decode_error` | Non-UTF-8 `plan.md` → `None` (was crashing) |
| `test_extract_session_name_no_h1_header` | `plan.md` with `## Subheading` only → `None` |
| `test_extract_session_name_empty_h1` | `plan.md` with `# \n` → `None` |

**`tests/copilot_usage/test_report.py`** — 3 new tests:
| Test | Covers |
|------|--------|
| `test_total_output_tokens_empty_metrics_includes_active_tokens` | Empty `model_metrics` + active tokens → active tokens |
| `test_total_output_tokens_pure_active_no_double_count` | Pure-active session doesn't double-count |
| `test_filter_sessions_excludes_session_with_no_start_time` | `start_time=None` always excluded |

## Verification

All 647 tests pass. Coverage: 99.35% (well above 80% threshold). Ruff, Pyright all clean.

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/23512105771) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `astral.sh`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "astral.sh"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

### CODE REVIEW

File: tests/copilot_usage/test_report.py [hunk 1]
```diff
@@ -2515,6 +2515,11 @@ def test_none_start_time_excluded_when_filtering(self) -> None:
         assert "no-time" not in ids
         assert "with-time" in ids
 
+    def test_filter_sessions_excludes_session_with_no_start_time(self) -> None:
+        since = datetime(2026, 1, 1, tzinfo=UTC)
+        session = SessionSummary(session_id="s", start_time=None)
+        assert _filter_sessions([session], since=since, until=None) == []
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_report.py]: Good catch — replaced the duplicate test with `test_none_start_time_included_when_no_bounds` as suggested, which covers the distinct behavior of sessions with `start_time=None` being included when no bounds are specified.

> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/23514271292)

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `astral.sh`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "astral.sh"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>
